test/functional/rpc_users.py

```diff
@@ -166,11 +189,18 @@ def run_test(self):
         self.stop_node(0)
 
         self.log.info('Check that failure to write cookie file will abort the node gracefully')
-        (self.nodes[0].chain_path / ".cookie.tmp").mkdir()
+        node0_cookie_file_path =     self.nodes[0].chain_path / ".cookie"
+        node0_cookie_file_path_tmp = self.nodes[0].chain_path / ".cookie.tmp"
+        node0_cookie_file_path_tmp.mkdir()
         self.nodes[0].assert_start_raises_init_error(expected_msg=init_error)
+        node0_cookie_file_path_tmp.rmdir()
```

Comment:
will the `node0_cookie_file_path_tmp` file be kept when assertion fails on purpose? If not, you could wrap it in a try/finally (deleting regardless of the assertion's outcome)

Reply:
Tests are running in a scratch directory - if part of the test fails, we don't worry about leaving some files behind.

Reply:
Can we remove the cleanup in that case?

Reply:
The cleanup is still there so that other tests following it in the Python script are not impacted.

Reply:
so do I understand it correctly that it's fine for other tests to be affected when this one already fails, leaving the directory?